### PR TITLE
Refactor SymbolTable and error handling

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
This PR modifies the symbol table interface to move some boilerplate code out of message handlers:

* adds `this(name: string)` access of symbol table
* adds `insert` method, which generates and returns a message response with the typical `"created" + st.attrib(rname)` information
* folds some error-handling into `executeCommand` to simplify the serverDaemon loop

### CODE REVIEW

**Review by ajpotts — APPROVED** [review]
Looks OK to me, but probably someone else should review the ServerDaemon.chpl changes more closely.

**Comment by jeremiah-corrado** [comment]
Sounds good, thanks! Also I'll likely have some more merge conflicts to resolve after #3358 gets merged, so others can feel free to hold off on reviewing until then.

**Comment by jeremiah-corrado** [comment]
This is fully ready for review whenever people have time, thanks!

File: src/ServerDaemon.chpl
Comment (stress-tess) [inline src/ServerDaemon.chpl:778]: it seems we duplicated this line, i'm not sure if that was intentional or not (I'm guessing not since the port didn't change

**Review by stress-tess — APPROVED** [review]
This looks good to me! I just have one question about a duplicated socket connect

File: src/ServerDaemon.chpl
Comment (jeremiah-corrado) [reply src/ServerDaemon.chpl:778]: Yeah not intentional, must have been some weirdness from resolving merge conflicts. Thanks for catching that!

**Review by stress-tess — APPROVED** [review]

**Pull request merged**
